lms/djangoapps/completion_api/views.py

```diff
@@ -84,10 +86,140 @@ def get_serializer(self):
 
 class CompletionListView(APIView, CompletionViewMixin):
     """
-    API view to render lists of serialized CourseCompletions.
+    API view to render serialized CourseCompletions for a single user
+    across all enrolled courses.
+
+    **Example Requests**
+
+        GET /api/completion/v0/course/
+        GET /api/completion/v0/course/?requested_fields=chapter,vertical
+
+    **Response Values**
+
+        The response is a dictionary comprising pagination data and a page
+        of results.
+
+        * pagination: A dict of pagination information, containing the fields:
+            * page: The page number of the current set of results.
+            * next: The URL for the next page of results, or None if already on
+              the last page.
+            * previous: The URL for the previous page of results, or None if
+              already on the first page.
+            * count: The total number of available results.
+        * results: A list of dictionaries representing the user's completion
+          for each course.
+
+        Standard fields for each completion dictionary:
+
+        * completion: A dictionary comprising of the following fields:
+            * earned (float): The sum of the learner's completions.
+            * possible (float): The total number of completions available
+              in the course.
+            * ratio (float in the range [0.0, 1.0]): The ratio of earned
+              completions to possible completions in the course.
+
+        Optional fields, as requested in "requested_fields":
+
+        * mean (float): The average completion ratio for all students enrolled
+          in the course.
+        * Aggregations: The actual fields available are configurable, but
+          may include `chapter`, `sequential`, or `vertical`.  If requested,
+          the field will be a list of all blocks of that type containing
+          completion information for that block.  Fields for each entry will
+          include:
+
+              * course_key (CourseKey): The unique course identifier.
+              * usage_key: (UsageKey) The unique block identifier.
+              * completion: A completion dictionary, identical in format to
+                the course-level completion dictionary.
+
+    **Parameters**
+
+        username (optional):
```

Comment:
@jcdyer I recently noticed an issue with the interim API: The docstrings, design spec, and course blocks API all use the parameter `username` for this, but the actual code is currently reading the `user` parameter, not `username`. It's not too late to change either the docs or the API itself - should we change it to `username` consistently?

Reply:
Gah.  Sorry I missed this comment.  

In the database, it's "user" when it is an actual link to the user model, and username when it's just a string.  I think it makes sense to keep it as username in the API, even if that is a mismatch with what's in the model, since the API deals with strings.